### DISCUSSION TITLE
Is this how I can change the name of the saved cropped file? (Yolov8)

### DISCUSSION THREAD

**Discussion by goodevile** [ultralytics/ultralytics#9326] [Q&A]
Thank you for your time.

The title is the question that I have.

in` /ultralytics/ultralytics/engine/predictor.py`, by changing the line below, is this how I can do it?

I'm using CLI: `yolo detect predict .............`

THIS:
```
if self.args.save_crop:
    result.save_crop(save_dir=self.save_dir / "crops", file_name=self.txt_path.stem)
```

To THIS:
```
if self.args.save_crop:
    result.save_crop(save_dir=self.save_dir / "crops", file_name = f"{self.txt_path.stem}_{datetime.now().strftime('%H%M%S%f')[:-3]}")
```

**Comment by glenn-jocher** [comment]
Yes, modifying that line in `predictor.py` as you've described is one way to change the naming convention of saved cropped files, appending the current time to the file name for uniqueness. Just ensure you import `datetime` at the top of your script:

```python
from datetime import datetime
```

Your modification looks good if you aim to include a timestamp in your filenames to avoid overwriting previous crops with the same name. Make sure to test this in a development environment before deploying it to production to ensure it behaves as expected. Happy coding! 😊

**Comment by goodevile** [comment]
thanks! it works.
![image](https://github.com/ultralytics/ultralytics/assets/46387951/7c2fe57d-a982-4c84-802d-121c037cb45c)

**Reply by pderrenger** [reply]
Fantastic! 🎉 I'm glad to hear it worked out for you. If you have any more questions or require further assistance, feel free to reach out. Happy coding with YOLOv8!
